src/security/PeerConnector.cc

```diff
@@ -640,27 +619,29 @@ Security::PeerConnector::certDownloadNestingLevel() const
 void
 Security::PeerConnector::startCertDownloading(SBuf &url)
 {
-    AsyncCall::Pointer certCallback = asyncCall(81, 4,
-                                      "Security::PeerConnector::certDownloadingDone",
-                                      PeerConnectorCertDownloaderDialer(&Security::PeerConnector::certDownloadingDone, this));
-
+    const auto certCallback = asyncCallback(81, 4, Security::PeerConnector::certDownloadingDone, this);
     const auto dl = new Downloader(url, certCallback,
         MasterXaction::MakePortless<XactionInitiator::initCertFetcher>(),
         certDownloadNestingLevel() + 1);
     certDownloadWait.start(dl, certCallback);
 }
 
 void
-Security::PeerConnector::certDownloadingDone(SBuf &obj, int downloadStatus)
+Security::PeerConnector::certDownloadingDone(DownloaderAnswer &downloaderAnswer)
 {
     certDownloadWait.finish();
 
+    // TODO: Remove diff reduction before merging
+    const auto &obj = downloaderAnswer.resource;
+    const auto downloadStatus = downloaderAnswer.outcome;
+
     ++certsDownloads;
     debugs(81, 5, "Certificate downloading status: " << downloadStatus << " certificate size: " << obj.length());
 
     Must(Comm::IsConnOpen(serverConnection()));
     const auto &sconn = *fd_table[serverConnection()->fd].ssl;
 
+    // XXX: Do not parse the response when the download has failed.
```

Comment:
Addressing this and other XXX/TODO PR-added comments (without "before merging" marks) are out of scope. If you do not think these comments should be added, please just say so, and I will remove them without arguing.